compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1998,7 +1998,24 @@ abstract class GenJSCode extends plugins.PluginComponent
           isRawJSCtorDefaultParam(sym)
         } else {
           sym.hasFlag(reflect.internal.Flags.DEFAULTPARAM) &&
-          isRawJSType(sym.owner.tpe)
+          isRawJSType(sym.owner.tpe) && {
+            /* If this is a default parameter accessor on a
+             * ScalaJSDefinedJSClass, we need to know if the method for which we
+             * are the default parameter is exposed or not.
+             * We do this by looking up the de-mangled name in the owner. Note
+             * that this does not work for local methods. But local methods are
+             * never exposed.
+             * Further note that overloads are easy, because either all or none
+             * of them are exposed.
+             */
+            def isOwnerMethodExposed = {
```

Comment:
"owner" is not very accurate wrt what scalac calls an owner. Consider maybe `isAttachedMethodExposed`?